src/main/java/com/exactpro/th2/conn/ServiceListener.java

```diff
@@ -123,20 +123,22 @@ public void exceptionCaught(IServiceProxy service, Throwable cause) {
 
     @Override
     public void onMessage(IServiceProxy service, IMessage message, boolean rejected, ServiceHandlerRoute route) {
-        LOGGER.debug("Handle message - route: {}; message: {}", route, message);
-        Direction direction = route.isFrom() ? FIRST : SECOND;
-        DIRECTION_TO_COUNTER.get(direction).labels(sessionAlias).inc();
-        AtomicLong directionSeq = directionToSequence.get(direction);
-        ConnectivityMessage connectivityMessage;
-
-        if (EvolutionBatch.MESSAGE_NAME.equals(message.getName())) {
-            EvolutionBatch batch = new EvolutionBatch(message);
-            connectivityMessage = createConnectivityMessage(batch.getBatch(), direction, directionSeq);
-        } else {
-            connectivityMessage = createConnectivityMessage(List.of(message), direction, directionSeq);
+        synchronized (subscriber) {
```

Comment:
We block both directions in this case. It is mean that incoming messages will have to wait for outgoing messages.
But we should execute make connectivity message and put it into the onNext message atomically

Reply:
So, we leave like this?